### PR TITLE
fix(controller): Fixes resource version misuse. Fixes #4714

### LINKED ISSUE #4714: Argo UI: Unknown error
## Summary 

When using the UI I randomly get:

```
Unknown error. Open your browser error console for more information.:  Reload 8s`
```
Reload button is dead.

Browser error console shows:

```
2:2746/api/v1/workflow-events/test-argo-2-12?listOptions.fieldSelector=metadata.name=create-6w7w2:1 GET http://internal-XXXXXXXXXXXXXXX.us-west-2.elb.amazonaws.com:2746/api/v1/workflow-events/test-argo-2-12?listOptions.fieldSelector=metadata.name=create-6w7w2 net::ERR_INCOMPLETE_CHUNKED_ENCODING 200 (OK)
```

## Diagnostics

What Kubernetes provider are you using? 

`AWS EKS v1.15.12-eks-31566f`

What version of Argo Workflows are you running? 

`v2.12.0-rc5`

OS and browser details:

Vivaldi | 3.4.2066.106 (Stable channel) (x86_64)
-- | --
Revision | 9faa7da6e287d30fe3948ac5d9d62aefe7977da5
OS | macOS Version 10.15.7 (Build 19H15)
JavaScript | V8 8.6.395.25

---

**Message from the maintainers**:

Impacted by this bug? Give it a 👍. We prioritise the issues with the most 👍.

### PR DESCRIPTION
**Pull request by alexec**
Signed-off-by: Alex Collins <alex_collins@intuit.com>

Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md).

### CODE REVIEW

File: cmd/argo/commands/wait.go [hunk 1]
```diff
@@ -74,7 +74,8 @@ func waitOnOne(serviceClient workflowpkg.WorkflowServiceClient, ctx context.Cont
 	req := &workflowpkg.WatchWorkflowsRequest{
 		Namespace: namespace,
 		ListOptions: &metav1.ListOptions{
-			FieldSelector: util.GenerateFieldSelectorFromWorkflowName(wfName),
+			FieldSelector:   util.GenerateFieldSelectorFromWorkflowName(wfName),
+			ResourceVersion: "0",
```
Comment (alexec) [inline cmd/argo/commands/wait.go:78]: "0" means, start from the latest version.

File: server/workflow/workflow_server.go [hunk 2]
```diff
@@ -210,13 +211,7 @@ func (s *workflowServer) WatchWorkflows(req *workflowpkg.WatchWorkflowsRequest,
 			return nil
 		case event, open := <-watch.ResultChan():
 			if !open {
-				log.Debug("Re-establishing workflow watch")
```
Comment (alexec) [inline server/workflow/workflow_server.go:213]: as the client must have re-connection logic, we should not try and mask/mitigate the problem here

File: ui/src/app/shared/services/requests.ts [hunk 3]
```diff
@@ -43,7 +43,22 @@ export default {
         return Observable.create((observer: Observer<any>) => {
             const eventSource = new EventSource(url);
             eventSource.onmessage = x => observer.next(x.data);
-            eventSource.onerror = x => observer.error(x);
+            eventSource.onerror = x => {
+                switch (eventSource.readyState) {
```
Comment (alexec) [inline ui/src/app/shared/services/requests.ts:47]: some users are creating "unknown error" issues - lets provide clearer error message

File: cmd/argo/commands/wait.go (hunk 1 above)
Comment (simster7) [reply cmd/argo/commands/wait.go:78]: Nifty trick

**Review by simster7 — APPROVED** [review]

**Pull request merged**